CMakeLists.txt

```
@@ -1454,12 +1454,27 @@ if(CONFIG_BUILD_OUTPUT_BIN)
 endif()
 
 if(CONFIG_BUILD_OUTPUT_BIN AND CONFIG_BUILD_OUTPUT_UF2)
+  if(CONFIG_BUILD_OUTPUT_UF2_IGNORE_FLASH_BASE)
+    set(FLASH_ADDR "0x0")
+  else()
+    set(FLASH_ADDR "${CONFIG_FLASH_BASE_ADDRESS}")
+  endif()
+
+  if(NOT CONFIG_BUILD_OUTPUT_UF2_IGNORE_FLASH_OFFSET)
+    # Note, the `+ 0` in formula below avoids errors in cases where a Kconfig
+    #       variable is undefined and thus expands to nothing.
+    math(EXPR FLASH_ADDR
+        "${FLASH_ADDR} + ${CONFIG_FLASH_LOAD_OFFSET} + 0"
+         OUTPUT_FORMAT HEXADECIMAL
+    )
+  endif()
+
   list(APPEND
     post_build_commands
     COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/uf2conv.py
             -c
             -f ${CONFIG_BUILD_OUTPUT_UF2_FAMILY_ID}
-            -b ${CONFIG_FLASH_LOAD_OFFSET}
+	    -b ${FLASH_ADDR}
```

I know it's a bit messy, but at least a function call split across multiple lines should be consistent internally.
```suggestion
           -b ${FLASH_ADDR}
```

---

Oof, yup, snuck in. Fixed.